python-package/lightgbm/sklearn.py

```diff
@@ -349,19 +337,14 @@ def fit(self, X, y,
             params['num_class'] = self.n_classes_
         if hasattr(self, 'eval_at'):
             params['ndcg_eval_at'] = self.eval_at
-
         if self.fobj:
-            params["objective"] = "None"
-        else:
-            params["objective"] = self.objective
+            params.pop('objective', None)
```

Comment:
@wxchan, the factory method of objective will return nullptr if passed string is not pre-defined. So we can use ```"None"``` to create a null objective.